examples/src/bin/autocomplete.rs

```diff
@@ -0,0 +1,66 @@
+use cursive::align::HAlign;
+use cursive::event::EventResult;
+use cursive::traits::*;
+use cursive::view::{Boxable, Identifiable};
+use cursive::views::{
+    Dialog, EditView, LinearLayout, OnEventView, SelectView, TextView,
+};
+use cursive::Cursive;
+use lazy_static::lazy_static;
+
+// This example shows a way to implement a (Google-like) autocomplete search box.
+// Try entering "Tok"!
+
+lazy_static! {
+    static ref CITIES: &'static str = include_str!("../../assets/cities.txt");
+}
+
+fn main() {
+    let mut siv = cursive::default();
+
+    siv.add_layer(
+        Dialog::around(
+            LinearLayout::vertical()
+                // the input is on the top
+                .child(EditView::new().on_edit(on_edit).with_name("query"))
+                // search results below the input
+                .child(
+                    SelectView::new()
+                        // shows all cities by default
+                        .with_all_str(CITIES.lines())
+                        // Sets the callback for when "Enter" is pressed.
+                        .on_submit(show_next_window)
+                        // Center the text horizontally
+                        .h_align(HAlign::Center)
+                        .with_name("matches"),
+                )
+                .fixed_width(25),
+        )
+        .button("Quit", Cursive::quit),
+    );
+
+    siv.run();
+}
+
+// Update results according to the query
+fn on_edit(siv: &mut Cursive, _content: &str, _cursor: usize) {
+    // Get the query
+    let query = siv.find_name::<EditView>("query").unwrap().get_content();
+    // Filter cities with names that starts with the query string
+    let matches = CITIES
+        .lines()
+        .filter(|&city| query.chars().zip(city.chars()).all(|(c, d)| c == d));
```

Comment:
Maybe [`str::starts_with`](https://doc.rust-lang.org/std/string/struct.String.html#method.starts_with) could help here?

Reply:
Also, maybe accepting more than just prefixes (using `contains()` instead) could help too?

Reply:
Thank you for the suggestion! I've changed it to `contains()`.